internal/services/v1/relationships_test.go

```diff
@@ -1122,6 +1122,40 @@ func TestWriteRelationshipsUpdatesOverLimit(t *testing.T) {
 	require.Contains(err.Error(), "update count of 2 is greater than maximum allowed of 1")
 }
 
+func TestReadRelationshipsWithTimeout(t *testing.T) {
+	require := require.New(t)
+
+	conn, cleanup, _, _ := testserver.NewTestServerWithConfig(
+		require,
+		0,
+		memdb.DisableGC,
+		false,
+		testserver.ServerConfig{
+			MaxUpdatesPerWrite:    1000,
+			MaxPreconditionsCount: 1000,
+			StreamingAPITimeout:   1,
+		},
+		tf.StandardDatastoreWithData,
+	)
+	client := v1.NewPermissionsServiceClient(conn)
+	t.Cleanup(cleanup)
+
+	stream, err := client.ReadRelationships(context.Background(), &v1.ReadRelationshipsRequest{
+		Consistency: &v1.Consistency{
+			Requirement: &v1.Consistency_FullyConsistent{FullyConsistent: true},
+		},
+		RelationshipFilter: &v1.RelationshipFilter{
+			ResourceType: "document",
+		},
+	})
+	require.NoError(err)
+	time.Sleep(1 * time.Millisecond)
+
+	// Ensure the recv fails with a context cancelation.
+	_, err = stream.Recv()
+	require.ErrorContains(err, "context canceled")
```

Comment:
`require.ErrorIs`?

Reply:
Do we know the type of the context cancelation error?

Reply:
unless we are doing something unorthodox with context cancellation, we should. `ErrorIs` should be able to unroll wrapped errors and identify the context cancellation type (which is a variable in the `context` package)